test/integration/xfcc_integration_test.cc

```diff
@@ -0,0 +1,243 @@
+#include "xfcc_integration_test.h"
```

Comment:
I haven't looked through this change yet, but, it seems unnecessary for a whole new config. Can it just be part of the SSL integration test? In general, the fact there are multiple different integration test configs is really arbitrary. They could all be in a single config which just has more listeners and more cluster definitions. I'm certainly happy to figure out  a way to simplify things there, but nothing comes to mind quickly, and that is orthogonal to this change. In this case I would probably just move the integration tests into the SSL one even if new listeners and upstreams need to be added.